toolchain/valgrind.py.data/valgrind_source.yaml

```diff
@@ -1,6 +1,7 @@
 run_type: !mux
     upstream:
-        url: 'ftp://sourceware.org/pub/valgrind/valgrind-3.18.1.tar.bz2'
+        url ="ftp://sourceware.org/pub/valgrind/"
+        version = self.params.get('version', default="3.18.1")
```

Comment:
@preetithakur the right way to add parameter to the yaml is as below.
version: ""

I think url is not required here as it is defined in .py itself and doesn't need any input from yaml.
So only above line is enough but make sure the indentation part.